VRDR.Messaging/BaseMessage.cs

```diff
@@ -192,6 +209,73 @@ public string MessageDestination
             }
         }
 
+        /// <summary>Jurisdiction-assigned death certificate number</summary>
+        public string CertificateNumber
+        {
+            get
+            {
+                return Record.GetSingleValue<FhirString>("cert_no")?.Value;
+            }
+            set
+            {
+                Record.Remove("cert_no");
+                if (value != null)
+                {
+                    Record.Add("cert_no", new FhirString(value));
+                }
+            }
+        }
+
+        /// <summary>Jurisdiction-assigned auxilliary identifier</summary>
+        public string StateIdentifier
+        {
+            get
+            {
+                return Record.GetSingleValue<FhirString>("state_id")?.Value;
+            }
+            set
+            {
+                Record.Remove("state_id");
+                if (value != null)
+                {
+                    Record.Add("state_id", new FhirString(value));
+                }
+            }
+        }
+
+        /// <summary>NCHS identifier. Format is 4-digit year, two character jurisdiction id, six character/digit certificate id.</summary>
+        public string NCHSIdentifier
+        {
+            get
+            {
+                return Record.GetSingleValue<FhirString>("nchs_id")?.Value;
+            }
+            set
+            {
+                Record.Remove("nchs_id");
+                if (value != null)
+                {
+                    Record.Add("nchs_id", new FhirString(value));
+                }
+            }
+        }
+
+        /// <summary>Create an NCHS identifier for the supplied DeathRecord.</summary>
+        /// <param name="record">the DeathRecord from which the year of death, jurisdiction of death and certificate id will be extracted.</param>
+        /// <returns>the NCHS compound identifier for the supplied DeathRecord.</returns>
+        protected static string CreateNCHSIdentifier(DeathRecord record)
+        {
+            var ije = new IJEMortality(record);
```

Comment:
There's quite a bit of logic in the IJE class to handle stuff like padding etc that is shared over multiple IJE fields. I think this logic belongs in the IJE class since it is related to the IJE fixed length field format which the NCHSIndetifier property inherits. Duplicating that code elsewhere could result in maintenance issues.